docs/proposals/all-in-one-volume.md

```diff
@@ -0,0 +1,258 @@
+## Abstract
+
+Describes a proposal for a new volume type that can project secrets,
+configmaps, and downward API items.
+
+## Motivation
+
+Users often need to build directories that contain multiple types of
+configuration and secret data. For example, a configuration directory for some
+software package may contain both config files and credentials. Currently, there
+is no way to achieve this in Kubernetes without scripting inside of a container.
+
+## Constraints and Assumptions
+
+1.  The volume types must remain unchanged for backward compatability.
+2.  There will be a new volume type for this proposed functionality, but no
+    other API changes.
+3.  The new volume type should support atomic updates in the event of an input
+    change.
+
+## Use Cases
+
+1.  As a user, I want to automatically populate a single volume with the keys
+    from multiple secrets, configmaps, and with downward API information, so
+    that I can synthesize a single directory with various sources of
+    information.
+2.  As a user, I want to populate a single volume with the keys from multiple
+    secrets, configmaps, and with downward API information, explicitly
+    specifying paths for each item, so that I can have full control over the
+    contents of that volume.
+
+A user should be able to map any combination of resources mentioned above into a
+single directory. There are plenty of examples of software that needs to be
+configured both with config files and secret data. The combination of having
+that data not only accessible, but in the same location provides for an easier
+user experience.
+
+In some cases, it may be easier to further define the path within the volume
```

Comment:
I would define this in terms of existing precedent and capabilities.  You need to be able to do everything in the new volume that you can in the old volumes.